gh_pages/_source/session8/Motion-Planning-with-Tesseract.rst

```diff
@@ -0,0 +1,431 @@
+
+   Now our TrajOpt planners are connected to our taskflow!
+
+#. Run the application:
+
+   Now try running the application again and notice how our robot's motion plan has changed. Also take a look at your ``.dot`` graph again to see the changes we have made to our taskflow. Don't forget to build and source your workspace!
+
+Implement the OMPL Planner Profile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Create the planner profile:
+   
+   Go back to ``planner_profiles.hpp`` and find the section
+
+   .. code-block:: c++
+
+      /* ======================
+       * Fill Code: OMPL
+       * ======================*/
+
+   OMPL is a libarary containing several different planning algorithms. OMPL allows us to use as many different planners in parallel as we'd like until one has a result. For our implementation, we will choose to use only RRT Connect. 
+
+   Below the above block, copy and past the following
+
+   .. code-block:: c++
+
+      auto n = static_cast<Eigen::Index>(std::thread::hardware_concurrency());
+      auto range = Eigen::VectorXd::LinSpaced(n, 0.005, 0.15);
+
+   This implements the number of threads we will have planning in parallel. Now we can add as many planners as available threads.
+
+   .. code-block:: c++
+
+      auto profile = std::make_shared<tesseract_planning::OMPLDefaultPlanProfile>();
+      profile->planning_time = 10.0;
+  	   profile->planners.reserve(static_cast<std::size_t>(n));
+	   for (Eigen::Index i = 0; i < n; ++i)
+	   {
+	     auto rrt_connect = std::make_shared<tesseract_planning::RRTConnectConfigurator>();
+	     rrt_connect->range = range(i);
+	     profile->planners.push_back(rrt_connect);
+	   }
+	   return profile;
+
+   There are many different OMPL planners available to experiment with. Feel free to play around with a few and observe how your application's motion plan changes (don't forget to include your chosen planner(s) in the header!).
+
+#. Add the planner to the planning server:
+
+   Return to ``planning_server.cpp`` and find the section where we add in our custom planning profiles.
+
+   Copy and paste the follwing
+
+   .. code-block:: c++
+
+      profile_dict_->addProfile<tesseract_planning::OMPLPlanProfile>(
+          tesseract_planning::profile_ns::OMPL_DEFAULT_NAMESPACE, PROFILE, createOMPLProfile());
+
+   Now we have added our new OMPL planning profile to the planning server.
+
+#. Add the planner to the taskflow:
+   
+   Go back to ``taskflow_generators.hpp``. Now we need to include our OMPL profile in our motion planning taskflow. Our OMPL planner will be used with TrajOpt in our freespace taskflow. First, let's create our node for OMPL. Within ``FreespaceMotionPipelineTask`` add
```

Comment:
We aren't directly including our OMPL profile here, we are just adding an OMPL task. In the createProgram method in the main source file we tell the motion planner to use a specific profile name and we created a profile under that identifiier. To actually use the profile we need to add an OMPL MotionPlannerTask here, but we could do that without creating a unique profile and it would just use the default settings.

Reply:
Also I wouldn't say "Our OMPL planner will be used with TrajOpt" I would say "Our freepsace taskflow will find a solution using OMPL and then improve that solution using TrajOpt" to make sure it is clear the OMPL and TrajOpt aren't completely independent.

Reply:
I see now, just below this you say almost exactly what my suggestion says. You can probably just move what you have below to here.